### PR TITLE
Added helm chart

### PR DESCRIPTION
**Pull request by njohnstone2**
Created a helm chart to support the cert manager install. If there's interest in getting this added to the repo I can support the other install options.

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
Thanks for the PR. 

I went through it and I guess it looks mostly good. Some things I think should be changed:
* The main branch has some modifications compared to the 0.3.1 release. So I think this needs to be adapted to it. For example the command and environment variables in the deployment changed.
* Ideally, I think it should with the default values give the same values as current installation files. That will make it much easier to test this properly. For example, it should create the namespace by default and use the same namespace as the regular installation files (i.e. `strimzi-drain-cleaner`). It is of course fine to make it configurable - just the defaults should correspond tot he regular installation files.
* It would be also great if it can in some of the important parts follow the configuration patterns from the operators Helm Chart. In particular, the image configuration comes to my mind here. In operators, this is configured through these fields: https://github.com/strimzi/strimzi-kafka-operator/blob/main/packaging/helm-charts/helm3/strimzi-kafka-operator/values.yaml#L11-L19
  This is again something what will significantly simplify the testing - and as nobody from the Strimzi core team really uses Helm, it is important to create a proper test coverage to make sure te Helm Chart keeps working.

I also (as someone who does not really use Helm) wonder how will this be published by release. Can this chart be somehow added to the same Helm repository where the operator is as well (which is currently https://strimzi.io/charts/index.yaml)?

**Comment by njohnstone2** [comment]
Thanks for the constructive feedback @scholzj. I believe i've addressed the main points in your comment. The labels change is inline with helm best practice but happy to update the Makefile to overwrite the `app` label to match the existing manifests (https://helm.sh/docs/chart_best_practices/labels/). 

Automating helm releases and azure pipelines are new territory for me so I likely haven't implemented everything required for the release. I'm happy to keep chipping away at this as I learn more or if you're not happy with the direction it's heading I won't be offended in abandoning the PR.

**Comment by scholzj** [comment]
Sorry, I did not get to this today, but I will have a look tomorrow.

**Comment by njohnstone2** [comment]
No rush! I likely won't be doing much on this until after new years. Enjoy the Christmas break!

File: packaging/helm-charts/helm3/Makefile
Comment (scholzj) [inline packaging/helm-charts/helm3/Makefile:41]: Is this really needed here given there are no CRDs?

**Review by scholzj — COMMENTED** [review]
Thanks for the changes. This looks mostly good to me. You actually did way more then I expected.

I left some comments - mostly about the labels. I do not mind the additional labels in the Helm Chart. But it would be great if it can add also the `app` label which we use (and keep it onthe plain YAML installation file). This is important as it makes it easy to find the things regardless how they were installed.

**Comment by scholzj** [comment]
> I also (as someone who does not really use Helm) wonder how will this be published by release. Can this chart be somehow added to the same Helm repository where the operator is as well (which is currently https://strimzi.io/charts/index.yaml)?

This is not something to be added to this PR, I can deal with the release process. I'm just looking for theoretical knowledge. Do you have an idea about how this works ^^^? (given you clearly understand Helm way better than I do ;-)). Can we just add this to the existing index.yaml and share the repository? Or will a separate one be needed?

**Comment by njohnstone2** [comment]
> Can we just add this to the existing index.yaml

Yes, a single helm repository can host multiple charts. You should be able to update the helm index the same way you do for the operator chart via the `helm repo index` command. I just tested locally by dropping the packaged helm chart (`.tgz` file) into the `charts/` directory of the [strimzi.github.io](https://github.com/strimzi/strimzi.github.io) project and updated the index.

e.g. 
```
cd src/github.com/strimzi/strimzi.github.io/charts

## Copy the packaged chart to the existing index.yaml location
cp /path/to/strimzi/drain-cleaner/strimzi-drain-cleaner-helm-3-chart-1.0.0.tgz .

## Update index.yaml and reference the v1.0.0 github release
helm repo index . --merge index.yaml --url https://github.com/strimzi/drain-cleaner/releases/download/1.0.0/

## Cleanup locally copied helm package
rm strimzi-drain-cleaner-helm-3-chart-1.0.0.tgz
```

Note that the above would need to be completed after the packaged helm chart is available in the github release for v1.0.0.

I have never needed to manage a static helm repo before (I personally use [chartmuseum](https://github.com/helm/chartmuseum) with S3 to manage persistence) which takes care of managing the state of `index.yaml`

File: packaging/install/certmanager/000-Namespace.yaml
Comment (njohnstone2) [reply packaging/install/certmanager/000-Namespace.yaml]: Fair point. I've updated the Makefile so the new labels only apply to the helm chart and the selector remains unchanged.

File: packaging/helm-charts/helm3/Makefile
Comment (njohnstone2) [reply packaging/helm-charts/helm3/Makefile]: This was a poor copy paste effort from the operator project. CRD logic has been removed.

**Review by im-konge — APPROVED** [review]
The changes LGTM, thanks for adding this :)

I will definitely add some test check for Helm installation method, I want to test it more manually in next few days :)

**Review by scholzj — APPROVED** [review]
Thanks for the changes. This looks good to me. I guess we should try to merge it to be able to start working on the tests etc. Thanks for the contribution!

**Comment by scholzj** [comment]
@ppatierno Do you plan to review this? With me and @im-konge, it now has the required approvals. We can wait if you plan to look at it. If you don't, I will proceed with merging it.

**Comment by ppatierno** [comment]
@scholzj I think you can move forward on this.

**Comment by scholzj** [comment]
Thanks for the PR @njohnstone2

**Pull request merged**
